oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -169,9 +176,23 @@ public void setDefaultClientRegistrationId(String clientRegistrationId) {
 	 * @return the {@link Consumer} to configure the builder
 	 */
 	public Consumer<WebClient.Builder> oauth2Configuration() {
+		Hooks.onLastOperator(REQUEST_CONTEXT_OPERATOR_KEY, Operators.lift((s, sub) -> createRequestContextSubscriber(sub)));
```

Comment:
We can make this implement DisposableBean to clean up the hooks

Reply:
Not sure I understand? The `ExchangeFilterFunction` would not really be a `@Bean`. The `WebClient` would be but that's not always the case either. What were you thinking exactly?

Reply:
We need a way to clean up the Hooks and a DisposableBean makes a lot of sense for that reason. Why wouldn't/couldn't it be a Bean? How would you clean it up?

Reply:
I agree `DisposableBean` is an option and yes we need to cleanup/release the Hook. I just don't understand who would be responsible for registering the `ExchangeFilterFunction` as a `@Bean`? Would this be required by the user to register? What if they don't register it? Am I missing something from what you are suggesting?

Reply:
The user is already in charge of creating the `ExchangeFilterFunction` and registering it with the `WebClient`. This just means they should create it as a `Bean`.

If they forget, then it is the same problem as any other `DisposableBean` that they didn't register as a Bean. Things end up in an inconsistent state. This scenario is really more important when the JVM process runs longer than the ApplicationContext (i.e. tests) I doubt it will cause a lot of issues, but it is not semantically correct to leave the Hook registered.

If you don't think DisposableBean is a good idea, what do you think we should do to ensure it gets cleaned up?

Reply:
> This just means they should create it as a `Bean`.

As a user I would question why do I need to register it as a `@Bean`?

> then it is the same problem as any other `DisposableBean` that they didn't register as a Bean

Yes but if the user is registering a user-defined `@Bean` and they have implemented a cleanup/release of resources than they know why they need to register it as a `@Bean`. With the `ExchangeFilterFunction`, the framework is doing the cleanup so it's not as obvious to them that they need to register it. It will have to be well documented so it's not missed.

> it is not semantically correct to leave the Hook registered.

Agreed, or else the `Context` will contain the request attributes for every sequence created which is not good.

We need to do something for sure. I'm thinking `PostConstruct` instead of `DisposableBean` since the framework recommends this approach as well. What do you think?